[vcpkg_acquire_msys.cmake] Add gettext tool

StarGate-One opened the pull request:
**Describe the pull request**
[vcpkg_acquire_msys.cmake] Add gettext tool needed by libpq

- What does your PR fix? Fixes #13291 [vcpkg_acquire_msys.cmake] missing download of gettext

- Which triplets are supported/not supported? Not Applicable

- Have you updated the CI baseline? Not Applicable

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)? Yes

StarGate-One committed:
Add tool gettext to vcpkg_acquire_msys.cmake

ras0219 commented on scripts/cmake/vcpkg_acquire_msys.cmake:
```diff
@@ -273,6 +273,10 @@ function(vcpkg_acquire_msys PATH_TO_ROOT_OUT)
     SHA512 f094a7f4926195ef7ba015f0c5c56587b1faa94d85530f07aaaa5557a1494c3bd75257d4687c8401cbf1328d23e5586a92b05f0a872caebb1a7e941a07829776
   )
+  msys_package(
+    URL "https://repo.msys2.org/msys/x86_64/gettext-0.19.8.1-1-x86_64.pkg.tar.xz"
```
This needs to list dependencies from https://packages.msys2.org/package/gettext?repo=msys&variant=x86_64 and be sorted into the list above in topological order; it looks like it needs `DEPS libasprintf libgettextpo libintl`.

StarGate-One replied:
@ras0219  This all beyond my technical abilities - I not msys expert - I see no topological order or any order alphabetical or otherwise or not instructions anywhere on how to modify the file. 
I went to page listed and see it list things that are in the download package already. 
I do not know why there was a change in direction from plain msys to downloading individual msys parts and pieces - msys with development packages would have everything needed. 
Since I not a technical as you all I withdraw and maybe the next person needing libpq and gettext is missing they can do better job or they smarter than me.

StarGate-One committed:
Updated vcpkg_acquire_msys.cmake per ras0219 instructions...
 1. Added dependencies to gettext-0.19.8.1-1-x86_64.pkg.tar.xz
    of libintl libgettextpo libasprintf
 2. Added dependency download for libgettextpo-0.19.8.1-1-x86_64.pkg.tar.xz
 3. Added dependency download for libasprintf-0.19.8.1-1-x86_64.pkg.tar.xz
 4. Put downloads in reverse dependcy order - seems it is called
    topological order now

ras0219 reviewed (approved):
Thanks for making the fixes and sorry for the confusing term! This is exactly what I meant :)

The reason we want to split up the msys2 directories is to avoid one port having a "hidden" dependency on another -- for example, when `A` installs gettext we don't want it to silently also be provided to `B`.

BillyONeal commented:
Thanks for your contribution!

StarGate-One commented:
> Thanks for your contribution!

You are most welcome @BillyONeal. Have a wonderful weekend... Thank you for all the hard work. 👍🏻
